src/Orleans/Utils/Pool/PooledResource.cs

```diff
@@ -54,7 +32,7 @@ public virtual void SignalPurge()
         /// </summary>
         public void Dispose()
         {
-            IObjectPool<T> localPool = Interlocked.Exchange(ref pool, null);
```

Comment:
We tend not to use 'var' unless the type is obvious, for readability reasons.  Type is none-obvious here.